File: AdvantageFramework.Core/EnumUtilities/Attributes/EnumObjectAttribute.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AdvantageFramework.Core.EnumUtilities.Attributes
{
    public partial class EnumObjectAttribute : Attribute
    {

        #region  Constants 



        #endregion

        #region  Enum 



        #endregion

        #region  Variables 

        private string _Code = "";
        private string _Description = "";

        #endregion

        #region  Properties 

        public string Code
        {
            get
            {
                string CodeRet = default;
                CodeRet = _Code;
                return CodeRet;
            }

            set
            {
                _Code = value;
            }
        }

        public string Description
        {
            get
            {
                string DescriptionRet = default;
                DescriptionRet = _Description;
                return DescriptionRet;
            }

            set
            {
                _Description = value;
            }
        }

        #endregion

        #region  Methods 

        public EnumObjectAttribute(string Code, string Description)
        {
            _Code = Code;
            _Description = Description;
        }

        public override string ToString()
        {
            string ToStringRet = default;
            ToStringRet = Code + " - " + Description;
            return ToStringRet;
        }

        #endregion

    }
}

File: AdvantageFramework/Enum Utilities/Attributes/EnumObjectAttribute.vb
Namespace EnumUtilities.Attributes

    <AttributeUsage(AttributeTargets.Field)> _
    Public Class EnumObjectAttribute
        Inherits Attribute

#Region " Constants "



#End Region

#Region " Enum "



#End Region

#Region " Variables "

        Private _Code As String = ""
        Private _Description As String = ""

#End Region

#Region " Properties "

        Public Property Code() As String
            Get
                Code = _Code
            End Get
            Set(ByVal value As String)
                _Code = value
            End Set
        End Property
        Public Property Description() As String
            Get
                Description = _Description
            End Get
            Set(ByVal value As String)
                _Description = value
            End Set
        End Property

#End Region

#Region " Methods "

        Public Sub New(ByVal Code As String, ByVal Description As String)

            _Code = Code
            _Description = Description

        End Sub
        Public Overrides Function ToString() As String

            ToString = Me.Code & " - " & Me.Description

        End Function

#End Region

    End Class

End Namespace

